src/midiEditorCommands.cpp

```diff
@@ -24,14 +25,51 @@ using namespace fmt::literals;
 
 // Note: while the below struct is called MidiControlChange in line with naming in Reaper,
 // It is also used for other MIDI messages.
-typedef struct {
+struct MidiControlChange{ 
 	int channel;
 	int index;
 	int message1;
 	int message2;
 	int message3;
 	double position;
-} MidiControlChange;
+	bool selected;
+	bool muted;
+
+	struct GetParams {
```

Comment:
I'm not a big fan of this GetParams thing for two reasons:

First, even though you're only using it when constructing the iterator, it will be available inside instances of the outer struct (e.g. MidiControlChange). Worse, the values in the instance of the outer struct will actually be wrong. I think this problem could be solved by moving this struct out and having the MidiEventIterator template take two arguments: the event type (e.g. MidiControlChange) and the requested params (e.g. MidiControlChangeReqParams).

Second, incantations like `MidiNoteIterator begin(take, MidiNote::GetParams{true, true, true, true, true})` aren't super readable. I guess we could comment the parameter names before each boolean (`/* position */ true`), but I wonder whether bit vectors would be easier and more readable:

```
enum class MidiControlChangeReqParams {
position = 1 << 0,
message = 1 << 1,
};
```

Then you could do:

`MidiNoteIterator begin(take, MidiControlChangeReqParams::position | MidiControlChangeReqParams::message1)`

MidiEventIterator's constructor would then just take an int with the bit vector. The only disadvantage there is that it's less type-safe - you could pass in CC flags for a note iterator - but I'm not sure how much of a real problem that is and the readability advantage might justify that sacrifice.

If you don't like the bit vector idea, I'm happy to go with the structs, but please move them out of the outer struct and comment the param names as described in my first point.

Reply:
> even though you're only using it when constructing the iterator, it will be available inside instances of the outer struct (e.g. MidiControlChange).

I'm so sorry. I'm completely wrong. For some reason, I was thinking this was defining the struct as a data member of the outer struct, but it isn't; it's just declaring a type, not a data member. This is totally fine. 😳